catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/CacheQueryFactory.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.cache.solr.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.opengis.filter.Filter;
+
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.source.solr.SchemaFields;
+
+public class CacheQueryFactory {
+
+    private FilterBuilder builder;
+
+    public CacheQueryFactory(FilterBuilder filterBuilder) {
+        builder = filterBuilder;
+    }
+
+    QueryRequest getQueryRequestWithSourcesFilter(QueryRequest input) {
+        QueryRequest queryWithSources = input;
+        if (input.getSourceIds() != null) {
```

Comment:
In general a query requests source ids can be null but probably not in the case of querying the cache. In the oddball case that it is null then queryWithSources variable will not be set in the if statement which is why I initialize it to the input and only have one return. Let me know if you want me to change this.